prep-source-build.sh

```diff
@@ -175,7 +175,23 @@ function DownloadArchive {
     archiveUrl="https://builds.dotnet.microsoft.com/source-built-artifacts/assets/Private.SourceBuilt.$archiveType.$archiveVersion.$archiveRid.tar.gz"
 
     echo "  Downloading source-built $archiveType from $archiveUrl..."
-    (cd "$packagesArchiveDir" && curl -f --retry 5 -O "$archiveUrl")
+    (
+      cd "$packagesArchiveDir" &&
+      for i in {1..5}; do
+        if curl -f --retry 5 -O "$archiveUrl"; then
+          exit 0
+        else
+          case $? in
+            18)
+              sleep 3
+              ;;
+            *)
+              exit 1
+              ;;
+          esac
+        fi
+      done
+    )
```

Comment:
I think we could simplify this by passing `-retry-delay 3 --retry-all-errors`

Reply:
Yes, I saw this option. Since there was a note in its [description](https://curl.se/docs/manpage.html#--retry-all-errors): `there may be unintended consequences such as sending or receiving duplicate data.`, I decided to use a for loop to handle the curl (18). But since you brought it up, I'll update the code now.

Reply:
Update: https://github.com/dotnet/dotnet/pull/1151/commits/0d9dd007641082d6794e1fc751333b6eecf74481

Reply:
yeah the docs mention that only happens when you redirect/pipe the curl output. since we explicitly use the -O option to let curl write the file we're save and it will handle deleting existing files